python/python/benchmarks/test_kmeans.py

```diff
@@ -22,7 +22,20 @@ def test_kmeans(benchmark):
     data = np.random.random((65535, 16)).astype("f")
 
     def _f():
-        kmeans = KMeans(256, "l2")
+        kmeans = KMeans(8192, "l2")
+        kmeans.fit(data)
+
+    benchmark(_f)
+
+
+@pytest.mark.benchmark(group="kmeans")
+def test_kmeans_torch(benchmark):
+    data = np.random.random((65535, 1536)).astype("f")
+
+    from lance.torch.kmeans import KMeans
+
+    def _f():
+        kmeans = KMeans(256, metric="cosine", device="cuda")
```

Comment:
I've never done anything with CUDA. I'll have to try this out on my desktop with NVIDIA GPU. :)

Reply:
ANother good thing about this is that, we can actually use macbook's GPU as well `device="mps"`